DotNet/Report/KafkaProducers/DataAcquisitionRequestedProducer.cs

```
@@ -47,8 +48,25 @@ public async Task<bool> Produce(ReportScheduleModel schedule, List<string>? pati
 
             foreach (string patientId in patientsToEvaluate)
             {
-                using var activity = new Activity("ReportService.ProduceDataAcquisitionRequested");
-                activity.Start();
+                // Generate the trace and span ID first
+                string traceId = ActivityTraceId.CreateRandom().ToHexString();
+                string spanId = ActivitySpanId.CreateRandom().ToHexString();
+                // Create a traceparent W3C format: version-traceId-spanId-flags
+                string traceparentValue = $"00-{traceId}-{spanId}-01";
+                // Create activity context from the generated IDs
+                var activityContext = new ActivityContext(
+                    ActivityTraceId.CreateFromString(traceId.AsSpan()),
+                    ActivitySpanId.CreateFromString(spanId.AsSpan()),
+                    ActivityTraceFlags.Recorded);
+
+
+                using var activity = ServiceActivitySource.Instance?.StartActivity(
```

Should throw an error if `ServiceActivitySource.Instance` is null?

---

logic extracted and retested.